test/Microsoft.AspNet.Mvc.Core.Test/Rendering/HtmlHelperSelectTest.cs

```diff
@@ -0,0 +1,533 @@
+                        "<option selected=\"selected\" value=\"1\">One</option>" + Environment.NewLine +
+                        "<option value=\"2\">Two</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"3\">Three</option>" + Environment.NewLine +
+                        "</optgroup>" + Environment.NewLine +
+                        "</select>"
+                    },
+                    {
+                        OneDisabledGroupAllSelectedSelectList,
+                        "<select id=\"Property1\" multiple=\"multiple\" name=\"Property1\">" +
+                        "<optgroup disabled=\"disabled\" label=\"Disabled Group\">" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"0\">Zero</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"1\">One</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"2\">Two</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"3\">Three</option>" + Environment.NewLine +
+                        "</optgroup>" + Environment.NewLine +
+                        "</select>",
+                        "<select id=\"Property1\" multiple=\"multiple\" name=\"Property1\">" +
+                        "<optgroup disabled=\"disabled\" label=\"Disabled Group\">" + Environment.NewLine +
+                        "<option value=\"0\">Zero</option>" + Environment.NewLine +
+                        "<option value=\"1\">One</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"2\">Two</option>" + Environment.NewLine +
+                        "<option value=\"3\">Three</option>" + Environment.NewLine +
+                        "</optgroup>" + Environment.NewLine +
+                        "</select>",
+                        "<select id=\"Property1\" multiple=\"multiple\" name=\"Property1\">" +
+                        "<optgroup disabled=\"disabled\" label=\"Disabled Group\">" + Environment.NewLine +
+                        "<option value=\"0\">Zero</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"1\">One</option>" + Environment.NewLine +
+                        "<option value=\"2\">Two</option>" + Environment.NewLine +
+                        "<option selected=\"selected\" value=\"3\">Three</option>" + Environment.NewLine +
+                        "</optgroup>" + Environment.NewLine +
+                        "</select>"
+                    },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(DropDownListDataSet))]
+        public void DropDownList_WithNullModel_GeneratesExpectedValue(
+            IEnumerable<SelectListItem> selectList,
+            string expectedHtml,
+            string ignoredHtml)
```

Comment:
see `DropDownList_WithModelValue_GeneratesExpectedValue()`, where the parameters are reversed.  did the same thing with other pairs and triples of `..._GeneratesExpectedValue*()` methods to keep the inputs together as much as possible.